lightning/src/util/events.rs

```diff
@@ -150,6 +150,23 @@ pub enum Event {
 		/// The outputs which you should store as spendable by you.
 		outputs: Vec<SpendableOutputDescriptor>,
 	},
+	/// This event is generated when a payment has been successfully forwarded through us and a
+	/// forwarding fee earned.
+	PaymentForwarded {
+		/// The fee, in milli-satoshis, which was earned as a result of the payment.
+		///
+		/// Note that if we force-closed a channel while a forwarded HTLC was pending, the amount
+		/// the next hop claimed may have been rounded down, causing a higher fee than expected.
+		///
+		/// If the channel which sent us the payment has been force-closed, we will claim the funds
+		/// via an on-chain transaction. In that case we do not consider the on-chain transaction
+		/// fees involved in such a claim and this value will be `None`. It is possible duplicate
+		/// `PaymentForwarded` events are generated for the same payment iff `fee_earned_msat` is
+		/// `None`.
+		fee_earned_msat: Option<u64>,
+		/// If this is `true`, the forwarded HTLC was claimed on-chain after a force-closure.
+		claim_from_onchain_tx: bool,
```

Comment:
s/force-closure/force-closure of the previous-hop channel/ or backwards channel?

Reply:
Let me know what you think of the new wording